### DISCUSSION TITLE
20230815

### DISCUSSION THREAD

**Discussion by Jaewui** [In-houseStudyGroup/Effective-Java-Study#12] [3. Q&A]
아이템1 정적 팩터리 메서드

- 장점 5 : 정적 팩터리 메서드를 작성하는 시점에는 반환할 객체의 클래스가 존재하지 않아도 된다 ??

**Comment by tjdxo1193** [comment]
박병우 주임의 발표
item1
이름을 가질 수 있다는게 장점이 아니지않나?
캐싱은 자료구조를 해쉬맵만 사용하나?

item2
룸북의 클래스위에 선언하는 @Bulider와 생성자에 선언하는것 차이?

item3
어디서 리플랙션은 어떻게 공격하나? 또 역직렬화는?

item6
유한 상태 기계-> 오토마타 (너무 지엽적임)
지연 초기화는 어떻게 구현되는지?
어댑터 패턴이 뭔가?

item7
스택을 이뮤터블하게 수정한 이유?
약한 참조는 어떻게 가비지 컬렉터에의해 구분되는가?
약한참조에 리터럴은 왜 gc의 대상이 되지 않을까?

**Comment by Jaewui** [comment]
**아이템1**
- JumboEnumSet, RegularEnumSet
- 캐싱에서 HashMap을 사용하는 이유

**아이템2**
- Builder에서 필수인 변수에 final을 붙이는 이유?
- @Builder 어노테이션에 클래스 레벨, 생성자 레벨, 메서드 레벨

**아이템4**
- Collections 가 무엇인가 ?
